examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1884 @@
+
+    // The left part of the mesh has a material ID of 0.
+    for (const auto &cell : tria_left.active_cell_iterators())
+      cell->set_material_id(0);
+
+    // The right face is non-matching. All other boundary IDs
+    // are set to 0.
+    for (const auto &face : tria_left.active_face_iterators())
+      if (face->at_boundary())
+        {
+          face->set_boundary_id(0);
+          if (face->center()[0] > 0.5 * length - 1e-6)
+            face->set_boundary_id(non_matching_id_left);
+        }
+
+    // Construct right part of mesh.
+    Triangulation<dim> tria_right;
+    const unsigned int subdiv_right = 1;
+    GridGenerator::subdivided_hyper_rectangle(tria_right,
+                                              {subdiv_right, 2 * subdiv_right},
+                                              {0.5 * length, 0.0},
+                                              {length, length});
+
+    // The right part of the mesh has a material ID of 1.
+    for (const auto &cell : tria_right.active_cell_iterators())
+      cell->set_material_id(1);
+
+    // The left face is non-matching. All other boundary IDs
+    // are set to 0.
+    for (const auto &face : tria_right.active_face_iterators())
+      if (face->at_boundary())
+        {
+          face->set_boundary_id(0);
+          if (face->center()[0] < 0.5 * length + 1e-6)
+            face->set_boundary_id(non_matching_id_right);
+        }
+
+    // Merge triangulations with tolerance 0 to ensure no vertices
+    // are merged.
+    GridGenerator::merge_triangulations(tria_left,
+                                        tria_right,
+                                        tria,
+                                        /*tolerance*/ 0.,
+                                        /*copy_manifold_ids*/ false,
+                                        /*copy_boundary_ids*/ true);
+    tria.refine_global(refinements);
+  }
+
+  // @sect3{Point-to-point interpolation}
+  //
+  // The main purpose of this function is to fill the remote communicator that
+  // is needed for point-to-point interpolation. Using this remote communicator
+  // also the corresponding remote evaluators are setup. Eventually, the
+  // operators are handed to the time integrator that runs the simulation.
+  //
+  template <int dim, typename Number>
+  void run_with_point_to_point_interpolation(
+    const MatrixFree<dim, Number>      &matrix_free,
+    const std::set<types::boundary_id> &non_matching_faces,
+    const std::map<types::material_id, std::pair<double, double>> &materials,
+    const double                                                   end_time,
+    const Function<dim> &initial_condition,
+    const std::string   &vtk_prefix)
+  {
+    const auto &dof_handler = matrix_free.get_dof_handler();
+    const auto &tria        = dof_handler.get_triangulation();
+    const auto &mapping     = *matrix_free.get_mapping_info().mapping;
+
+    // Communication objects know about the communication pattern. I.e.,
```

Comment:
I am wondering if we should have a set of `Utility` functions that set up commonly used use cases of `FERemoteEvaluationCommunicator`? Such a function could be called here.